addons/hr_expense/models/hr_expense.py

```diff
@@ -163,47 +162,37 @@ def _compute_state(self):
             else:
                 expense.state = "done"
 
-    @api.depends('quantity', 'unit_amount', 'tax_ids', 'currency_id')
+    @api.depends('quantity', 'unit_amount', 'tax_ids')
     def _compute_amount(self):
         for expense in self:
             if expense.product_id and not expense.unit_amount:
                 continue
-            taxes = expense._get_taxes(price=expense.unit_amount, quantity=expense.quantity)
+            taxes = expense._get_taxes_results(expense.unit_amount, expense.quantity, expense.currency_id)
             expense.total_amount = taxes['total_included']
 
-    @api.depends('total_amount', 'tax_ids', 'currency_id')
+    @api.depends('total_amount', 'currency_rate')
     def _compute_amount_tax(self):
         """Note: as total_amount can be set directly by the user (for product without cost) or needs to be computed (for product with cost),
            `untaxed_amount` can't be computed in the same method as `total_amount`.
         """
         for expense in self:
-            taxes = expense._get_taxes(price=expense.total_amount, quantity=1.0)
+            taxes = expense._get_taxes_results(expense.total_amount, 1.0, expense.currency_id)
```

Comment:
`_get_taxes_results` will be called twice in some cases - can we not avoid that?
I guess the docstring explains why